ziti/cmd/verify/ops_verify_network.go

```diff
@@ -0,0 +1,263 @@
+/*
+	Copyright NetFoundry Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+package verify
+
+import (
+	"fmt"
+	"gopkg.in/yaml.v3"
+	"net"
+	"os"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/openziti/ziti/common"
+)
+
+type network struct {
+	controller
+	controllerConfig string
+	routerConfig     string
+	verbose bool
+}
+
+func NewVerifyNetwork() *cobra.Command {
+	n := &network{}
+
+	cmd := &cobra.Command{
+		Use:   "verify-network",
+		Short: "Verifies the overlay is configured correctly",
+		Long:  "A tool to verify network configurations, checking controller and router ports or other common problems",
+		Run: func(cmd *cobra.Command, args []string) {
+			if n.verbose {
+				log.SetLevel(log.DebugLevel)
+			}
+			configureLogFormat()
+			anyFailure := false
+			if n.controllerConfig != "" {
+				log.Infof("Verifying controller config: %s", n.controllerConfig)
+				anyFailure = verifyControllerConfig(n.controllerConfig) || anyFailure
+				fmt.Println()
+			}
+			if n.routerConfig != "" {
+				log.Infof("Verifying router config: %s", n.routerConfig)
+				anyFailure = verifyRouterConfig(n.routerConfig) || anyFailure
+				fmt.Println()
+			}
+			if anyFailure {
+				log.Error("One or more error. Review the output above for errors.")
+			} else {
+				log.Info("All requested checks passed.")
+			}
+		},
+	}
+
+	cmd.Flags().StringVarP(&n.controllerConfig, "controller-config-file", "c", "", "the controller config file verify")
+	cmd.Flags().StringVarP(&n.routerConfig, "router-config-file", "r", "", "the router config file to verify")
+	cmd.Flags().BoolVar(&n.verbose, "verbose", false, "Show additional output.")
+
+	return cmd
+}
+
+type StringMap map[string]interface{}
+
+func (m StringMap) mapFromKey(key string) StringMap {
+	if v, ok := m[key]; ok {
+		return v.(StringMap)
+	}
+	log.Fatalf("map didn't contain key %s", key)
+	return nil
+}
+
+type StringMapList []interface{}
+
+func (m StringMap) listFromKey(key string) StringMapList {
+	if v, ok := m[key]; ok {
+		return v.([]interface{})
+	}
+	log.Fatalf("map didn't contain key %s", key)
+	return nil
+}
+
+func (p ProtoHostPort) testPort(msg string) bool {
```

Comment:
Style choice, but these usually appear after the type definition.

Reply:
moved types to the top